operations/helm/charts/mimir-distributed/values.yaml

```diff
@@ -990,133 +990,349 @@ compactor:
   extraEnvFrom: []
 
 memcached:
+  # -- Specifies whether memcached should be enabled
   enabled: false
-  architecture: high-availability
-  arguments:
-    - -m 8192
-    - -o
-    - modern
-    - -v
-    - -I 1m
-    - -c 4096
+
+  # -- Total number of memcached replicas
+  replicas: 1
+
+  # -- Container image settings for memcached
   image:
     repository: memcached
-    tag: 1.6.9
-  # maxItemMemory is in bytes. Should match memcached -I flag (which is in MB)
-  # It is a string to avoid https://github.com/helm/helm/issues/1707.
-  maxItemMemory: '1048576'  # (* 1 (* 1024 1024))
-  metrics:
-    enabled: true
-    image:
-      registry: quay.io
-      repository: prometheus/memcached-exporter
-      tag: v0.9.0
-  replicaCount: 1
+    tag: 1.6.9-alpine
+    pullPolicy: IfNotPresent
+
+  # -- Port of the memcached service
+  port: 11211
+
+  # -- Amount of memory allocated to memcached for object storage (in MB).
+  allocatedMemory: 8192
+
+  # -- Maximum item memory for memcached (in MB).
+  maxItemMemory: 1
+
+  # -- Extra init containers for memcached pods
+  initContainers: []
+  # -- Security context for memcached pods
+  securityContext: {}
+  # -- Annotations for the memcached pods
+  annotations: {}
+  # -- Node selector for memcached pods
+  nodeSelector: {}
+  # -- Affinity for memcached pods
+  affinity: {}
+  # -- Tolerations for memcached pods
+  tolerations: []
+  # -- Pod Disruption Budget
+  podDisruptionBudget: {}
+  # -- The name of the PriorityClass for memcached pods
+  priorityClassName: null
+  # -- Labels for memcached pods
+  podLabels: {}
+  # -- Annotations for memcached pods
+  podAnnotations: {}
+  # -- Management policy for memcached pods
+  podManagementPolicy: Parallel
+  # -- Grace period to allow the memcached to shutdown before it is killed
+  terminationGracePeriodSeconds: 60
+
+  # -- Stateful memcached strategy
+  statefulStrategy:
+    type: RollingUpdate
+
+  # -- Additional CLI args for memcached
+  extraArgs: []
+
+  # -- Additional containers to be added to the memcached pod.
+  extraContainers: []
+
+  # -- Resource requests and limits for the memcached
   resources:
     limits:
       # memory limits should match requests
       memory: 9830Mi
     requests:
       cpu: 500m
```

Comment:
I was more addressing the fact that there are two implicitly tied configuration parameters. the `-m` flag and the pod resources. if they should always be the same is there a point in having two options? Maybe I'm missing the use case where they should _not_ be the same.

Another thing - I think the comment should be updated to mention the `allocatedMemory` config option instead of the `-m` flag. We used to expose the `-m` flag, but we replaced it with `allocatedMemory` now.